File: stacks/ltcd_2334.py
'''
LeetCode - problem 2334
'''

    def validSubarraySize(self, nums: List[int], threshold: int) -> int:
        nums.append(0)
        stack = [-1]
        for i in range(len(nums)):
            while nums[stack[-1]]>nums[i]:
                w = i-stack[-2]-1
                if nums[stack.pop()] > threshold / w:
                    return w
            stack.append(i)
        return -1

File: stacks/ltcd_2334.cpp
//LeetCode - problem 2334

    int validSubarraySize(vector<int>& nums, int threshold) {
        int n = nums.size();
        stack<int> st;
        for (int i = 0; i <= n; ++i) {
            while (!st.empty() and (i == n or nums[st.top()] > nums[i])) {
                int j = st.top(); st.pop(); 
                int left = st.empty() ? -1 : st.top(); 
                int len = i-left-1;
                double t = threshold*1.0/len*1.0;
                if (nums[j] > t) return len;
            }
            st.push(i);
        }
        return -1;
    }
